src/datasources/balances-api/entities/zerion-collectible.entity.ts

```diff
@@ -1,40 +1,77 @@
-export interface ZerionCollectionInfo {
-  content: {
-    icon: { url: string };
-    banner: { url: string };
-  } | null;
-  description: string | null;
-  name: string | null;
-}
-
-export interface ZerionNFTInfo {
-  content: {
-    preview: { url: string } | null;
-    detail: { url: string } | null;
-  } | null;
-  contract_address: string;
-  flags: { is_spam: boolean } | null;
-  interface: string | null;
-  name: string | null;
-  token_id: string;
-}
-
-export interface ZerionCollectibleAttributes {
-  amount: string;
-  changed_at: string;
-  collection_info: ZerionCollectionInfo | null;
-  nft_info: ZerionNFTInfo;
-  price: number;
-  value: number;
-}
-
-export interface ZerionCollectible {
-  attributes: ZerionCollectibleAttributes;
-  id: string;
-  type: 'nft_positions';
-}
-
-export interface ZerionCollectibles {
-  data: ZerionCollectible[];
-  links: { next: string | null };
-}
+import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import { z } from 'zod';
+
+export type ZerionCollectionInfo = z.infer<typeof ZerionCollectionInfoSchema>;
+
+export type ZerionNFTInfo = z.infer<typeof ZerionNFTInfoSchema>;
+
+export type ZerionCollectibleAttributes = z.infer<
+  typeof ZerionCollectibleAttributesSchema
+>;
+
+export type ZerionCollectible = z.infer<typeof ZerionCollectibleSchema>;
+
+export type ZerionCollectibles = z.infer<typeof ZerionCollectiblesSchema>;
+
+const ZerionCollectionInfoSchema = z.object({
+  content: z
+    .object({
+      icon: z.object({
+        url: z.string(),
+      }),
+      banner: z.object({
+        url: z.string(),
+      }),
+    })
+    .nullable(),
+  description: z.string().nullable(),
+  name: z.string().nullable(),
+});
+
+const ZerionNFTInfoSchema = z.object({
+  content: z
+    .object({
+      preview: z
+        .object({
+          url: z.string(),
+        })
+        .nullable(),
+      detail: z
+        .object({
+          url: z.string(),
+        })
+        .nullable(),
+    })
+    .nullable(),
+  contract_address: AddressSchema,
+  flags: z
+    .object({
+      is_spam: z.boolean(),
+    })
+    .nullable(),
+  interface: z.string().nullable(),
+  name: z.string().nullable(),
+  token_id: z.string(),
+});
+
+const ZerionCollectibleAttributesSchema = z.object({
+  amount: z.string(),
+  changed_at: z.string(), // Coerce?
+  collection_info: ZerionCollectionInfoSchema.nullable(),
```

Comment:
I'd say we can coerce this, but it's not used anywhere yet, so feel free to leave it as it is if you prefer.

Reply:
I forgot to remove this working note 🙈 Considering it's not used, I coerced it in 78a9197b4cd143e8e73bc060c66ed657275d8a04 for future use.